src/Equinox.Cosmos/Cosmos.fs

```diff
@@ -1109,28 +1123,38 @@ type Resolver<'event, 'state, 'context>(context : Context, codec, fold, initial,
         | CachingStrategy.SlidingWindow(cache, window) ->
             Caching.applyCacheUpdatesWithSlidingExpiration cache null window folder
 
-    let resolveStream (streamId, maybeContainerInitializationGate) opt context =
+    let resolveStream (streamId, maybeContainerInitializationGate) opt context compress =
         { new IStream<'event, 'state> with
             member __.Load log = category.Load(log, streamId, opt)
             member __.TrySync(log: ILogger, token: StreamToken, originState: 'state, events: 'event list) =
                 match maybeContainerInitializationGate with
-                | None -> category.TrySync(log, token, originState, events, context)
+                | None -> category.TrySync(log, token, originState, events, context, compress)
                 | Some init -> async {
                     do! init ()
-                    return! category.TrySync(log, token, originState, events, context) } }
+                    return! category.TrySync(log, token, originState, events, context, compress) } }
 
     let resolveTarget = function
         | StreamName.CategoryAndId (categoryName, streamId) -> context.ResolveContainerStream(categoryName, streamId)
 
-    member __.Resolve(streamName : StreamName, [<O; D null>]?option, [<O; D null>]?context) =
+    member __.Resolve
+        (   streamName : StreamName,
+            [<O; D null>]?option,
+            [<O; D null>]?context,
+            /// Determines whether the data and metadata payloads of the `u`nfolds in the Tip document are base64 encoded and compressed; defaults to true
+            [<O; D true>]?compress) =
```

Comment:
should this simply be called `compressSnapshots` or `compressUnfolds` to remove the ambiguity and/or need for xmldoc ? (does not have to have the longer name all the way inwards)

Reply:
Sounds reasonable, renamed.